nntrainer/tensor/manager.cpp

```diff
@@ -286,6 +287,8 @@ void Manager::deallocateWeights() {
       weight.deallocateVariable();
     }
   }
+
+  weights_allocated = false;
```

Comment:
weights_initialized = false; ??

Reply:
This is a bit tricky. 
weight_allocated and weight_initialized variables are not synced. It is just that `initializeWeights()` does allocation as well - this can be separated as well, but there is not advantage to it and delays loading the saved model file.
Weights are still initialized (with correct configuration) after dealloc. And can be allocated again with initialization.